building-with-ocean/asset-hosting.md

```diff
@@ -17,134 +17,136 @@ On Ocean Marketplace, a publisher must provide the link to the asset during publ
+In order to publish an asset on the market, the S3 URL needs to be public.
+This step shows how to set up access control policies to grant permissions to others. 
 
-**Step 2 - Create a blob container**
+**Editing permissions**
 
-![Create a blob container](../.gitbook/assets/azure-5.png)
+Go to the `Permissions` tab and select `Edit` and then uncheck `Block all public access` 
+boxes to give everyone read access to the object and click `Save`.
 
-**Step 3 - Upload a file**
+If editing the permissions is unavailable, modify the `Object Ownership` by enabling the ACLs
+as shown below.
 
-![Upload a file](../.gitbook/assets/azure-6.png)
+![Access the Object URL on S3 Bucket - 1](../.gitbook/assets/aws-9.png)
 
-**Step 4 - Share the file**
+**Modifying bucket policy**
 
-**Select the file to be published and click Generate SAS**
+In order to have the bucket granted on public access, its policy needs to be
+modified likewise.
 
-![Click generate SAS](../.gitbook/assets/azure-7.png)
+Note that the `<BUCKET-NAME>` must be chosen from personal buckets' dashboard.
 
-**Configure the SAS details and click `Generate SAS token and URL`**
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "Public S3 Bucket",
+            "Principal": "*",
+            "Effect": "Allow",
+            "Action": "s3:GetObject",
+            "Resource": "arn:aws:s3:::<BUCKET-NAME>/*"
+        }
+    ]
+}
+```
 
-![Generate link to file](../.gitbook/assets/azure-8.png)
+After saving the changes, the bucket should appear as `Public` access.
 
-**Copy the generated link**
+![Access the Object URL on S3 Bucket - 2](../.gitbook/assets/aws-10.png)
 
-![Copy the link](../.gitbook/assets/azure-9.png)
+**Verify the object URL on public access**
 
-**Step 5 - Publish the asset using the generated link**
+Select the file from the bucket that needs verification and select `Open`. Now it should
+be able to download the file on personal system.
 
-Now, copy and paste the link in the Publish page in the Ocean Marketplace.
+![Access the Object URL on S3 Bucket - 3](../.gitbook/assets/aws-11.png)
 
-![Publish the file as an asset](../.gitbook/assets/azure-10.png)
 
-#### OneDrive
+**Step 4 - Get the S3 Bucket Link & Publish Asset on Market**
 
-Create an account on [Microsoft](https://www.microsoft.com/en-us/microsoft-365/onedrive/online-cloud-storage).
+Now that the S3 endpoint has public access, the asset will be hosted successfully.
 
-**Step 1 - Upload a file**
+Go to [Ocean Market](https://market.oceanprotocol.com/publish/1) to complete the form
+for asset creation.
 
-Go to [OneDrive](https://onedrive.live.com/) and upload the file to be published.
+Copy the `Object URL` that can be found at `Object Overview` from AWS S3 bucket
+and paste it in the `File` field from the form as it is illustrated below.
 
-![Upload a file](../.gitbook/assets/one-drive-1.png)
+![Get the S3 Bucket Link & Publish Asset on Market - 1](../.gitbook/assets/aws-12.png)
```

Comment:
We can have a bigger screenshot here? To reflect the page to the user. Also, we usually use dark mode for market screenshots. If you can do the same, it would be great

Reply:
I will add this change in a separate commit. Thank you @AnaLoznianu for the review!